microservice/challenge-2/src/main/java/com/byond/savethehumans/dispatch/Dispatcher.java

```diff
@@ -0,0 +1,121 @@
+package com.byond.savethehumans.dispatch;
+
+import com.byond.savethehumans.learner.WordToVector;
+import com.byond.savethehumans.model.ContentResponse;
+import com.byond.savethehumans.model.UserReview;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+/**
+ *
+ * Dispatcher holds the business logic of detecting bot/human reviewers.
+ *
+ */
+
+@ComponentScan
+@Component
+@Service
+public class Dispatcher {
+
+
+    protected static final Logger logger = LoggerFactory.getLogger(Dispatcher.class);
+
+
+    @Autowired
+    private  WordToVector w2v;
+
+    @Value("${wordDistance}")
+    private int wordDistance;
+
+    @Value("${wordToVector.enabled}")
+    private Boolean word2VecIsEnabled;
+
+
+    /*  Loads the default suspected words such as Amazing and Human */
+    @Value("#{'${defaultSuspectedWords}'.split(',')}")
+    private  Set<String> defaultSuspectedWords;
+
+    Set<String> allSuspectedWords = new HashSet<>();
+
+    public ContentResponse investigate(UserReview c) throws Exception {
+
+        if (!word2VecIsEnabled) {
```

Comment:
Why would word2Vec not be enabled? The application can't run without it...

Reply:
Good question. There was a Multiple-Classifier as well. Just beside W2V. I wanted to use it beside W2V to make more exciting results. In the contentResponse DTO, we had a set of tags that were coming from the classifier such as Automotive, Video, Music ... 
At that time I had:
```
word2VecIsEnabled=true
reviewClassifier=true
```
So I was able to make it like this:
```
word2VecIsEnabled=false
reviewClassifier=true
```
and train the classifier faster...
When I noticed the results aren't enough good, I just removed the classifier.
word2VecIsEnabled also can be removed.